components/channel_members_rhs/member.tsx

```diff
@@ -102,32 +104,66 @@ interface Props {
     };
 }
 
+interface MMOverlayTrigger extends BaseOverlayTrigger {
+    hide: () => void;
+}
+
 const Member = ({className, channel, member, index, totalUsers, editing, actions}: Props) => {
+    const overlay = React.createRef<MMOverlayTrigger>();
+    const profileSrc = Client4.getProfilePictureUrl(member.user.id, member.user.last_picture_update);
```

Comment:
0/5 but we can put this in useEffect with member.user.id as dependency. It will not allow this to be called multiple times. Although this wont be a problem because browser will not allow this through as it will be cached.

Reply:
Great idea, done!

Reply:
`getProfilePictureUrl` is just a string builder (no request is made to the server), I'm not sure introducing the complexity of useEffect here is worth it (probably much heavier than rebuilding the string every time)

Reply:
https://github.com/mattermost/mattermost-webapp/blob/master/packages/client/src/client4.ts#L939-L947

Reply:
Well the user ID is static for a given user and the URL for a profile picture is based on this id, so this will likely never change. If the component rerenders, it will be because for something different. I just think the code induced by calling `getProfilePictureUrl` is negligeable and whenever possible, I like the code to look simple :P

Reply:
Okay, I rolled this back to the original version without the `useEffect`